File: Max_subarray/code.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans = INT_MIN;
        int present = 0;
        for(int i =0;i<nums.size();i++){
            present += nums[i];
            if( ans < present) ans = present;
            if(present < 0) present = 0;
        }
        return ans;
    }
};

#pragma GCC optimize("Ofast")
static auto _ = [](){
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);
    std::cout.tie(nullptr);
    return nullptr;
}();

File: Max_subarray/code.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_so_far = float('-inf')

        max_end = 0
        for i in nums:
            max_end = max_end + i
            if max_end > max_so_far:
                max_so_far = max_end
            if max_end < 0:
                max_end = 0
        return max_so_far

                




        

        